pkgs/os-specific/linux/decklink/default.nix

```
@@ -0,0 +1,49 @@
+{ stdenv, lib, requireFile, fetchpatch, kernel }:
+
+stdenv.mkDerivation rec {
+  pname = "decklink";
+  major = "12.2";
+  version = "${major}a12";
+
+  src = requireFile {
+    name = "Blackmagic_Desktop_Video_Linux_${major}.tar.gz";
+    url = "https://www.blackmagicdesign.com/support/download/33abc1034cd54cf99101f9acd2edd93d/Linux";
+    sha256 = "62954a18b60d9040aa4a959dff30ac9c260218ef78d6a63cbb243788f7abc05f";
```

```suggestion
    sha256 = "62954a18b60d9040aa4a959dff30ac9c260218ef78d6a63cbb243788f7abc05f";
    message = ''
      Please download the file manually from the developers website.
      They want you to register your product, but you can also download
      the file without registration by clicking 'Download Only' in the
      bottom left corner of the dialog.
      Then add the downloaded file to the Nix store using:
      nix-store --add-fixed sha256 Blackmagic_Desktop_Video_Linux_12.2.tar.gz
    '';
```

I'm not sure if you have to include the command there. But those instructions seem helpful.

---

In https://github.com/NixOS/nixpkgs/pull/152113/files is a script to download files from blackmagicdesign.com. Maybe that would work here too.

---

It indeed works. Thanks for the pointer, I was finally able to implement that 1.5 years later...